lychee-lib/src/extract/html.rs

```diff
@@ -28,25 +28,24 @@ impl TokenSink for LinkExtractor {
                 } = tag;
 
                 for attr in attrs {
-                    let urls = extract_urls_from_elem_attr(
+                    let urls = LinkExtractor::extract_urls_from_elem_attr(
                         attr.name.local.as_ref(),
                         name.as_ref(),
                         attr.value.as_ref(),
                     );
 
-                    if urls.is_empty() {
-                        extract_plaintext(&attr.value);
-                    } else {
-                        self.links.extend(
-                            urls.into_iter()
-                                .map(|url| RawUri {
-                                    text: url,
-                                    element: Some(name.to_string()),
-                                    attribute: Some(attr.name.local.to_string()),
-                                })
-                                .collect::<Vec<_>>(),
-                        );
-                    }
+                    let new_urls = match urls {
+                        None => extract_plaintext(&attr.value),
+                        Some(urls) => urls
+                            .into_iter()
+                            .map(|url| RawUri {
+                                text: url.to_string(),
```

Comment:
to_string() is available for types implementing Display.

There's no convention about whether Display gives same output across versions.

So it's always better to use String::from() when available.